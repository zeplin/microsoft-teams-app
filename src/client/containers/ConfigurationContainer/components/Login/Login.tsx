import React, { FunctionComponent } from "react";
import { Button, Flex, Text, ErrorIcon } from "@fluentui/react-northstar";
import { ZEPLIN_WEB_APP_BASE_URL } from "../../../../lib/config";

interface LoginProps {
    error?: string;
    onButtonClick: () => void;
}

const signupURL = new URL(ZEPLIN_WEB_APP_BASE_URL);
signupURL.pathname = `signup`;

export const Login: FunctionComponent<LoginProps> = ({ error, onButtonClick }) => (
    <Flex fill column gap="gap.medium">
        <Flex fill column gap="gap.large">
            <div />
            <Flex fill column gap="gap.medium">
                <Text size="large" weight="semibold">
                    One more step before finishing connection
                </Text>
                <Text>
                    Once proceeding, we will ask you to authorize Microsoft Teams access through a separate window. Once
                    it’s authorized, you can do more advanced settings on the Microsoft Teams channel linked with
                    Zeplin. If you’re not a Zeplin user, you can sign up from <Text
                        as="a"
                        color="brand"
                        href={signupURL.toString()}
                        target="_blank"
                        styles={{
                            "textDecoration": "none",
                            ":hover": {
                                textDecoration: "underline"
                            }
                        }}>
                        here
                    </Text>.
                </Text>
                {error && (
                    <Flex fill gap="gap.smaller">
                        <ErrorIcon size="large" />
                        <Text error>
                            {error}
                        </Text>
                    </Flex>
                )}
            </Flex>
            <div>
                <Button
                    primary
                    content="Log in Zeplin"
                    onClick={(): void => onButtonClick()} />
            </div>
        </Flex>
        <Text>
        You can learn more about the integration from <Text
                as="a"
                color="brand"
                href="https://zpl.io/msteams-integration-help"
                target="_blank"
                styles={{
                    "textDecoration": "none",
                    ":hover": {
                        textDecoration: "underline"
                    }
                }}>
                here
            </Text>.
        </Text>
    </Flex>
);
