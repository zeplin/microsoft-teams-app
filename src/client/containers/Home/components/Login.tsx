import React, { FunctionComponent } from "react";
import { Button, Flex, Text } from "@fluentui/react-northstar";
import * as microsoftTeams from "@microsoft/teams-js";
import { BASE_URL } from "../../../config";

interface LoginProps {
    onTokenReceive: (token: string) => void;
}

export const Login: FunctionComponent<LoginProps> = ({ onTokenReceive }) => (
    <Flex fill column gap="gap.small" styles={{ padding: "60px" }}>
        <Text size="large" content="One more step before finishing connection" weight="semibold" />
        <Flex fill column gap="gap.medium">
            <Text>
                Once proceeding, we will ask you to authorize Microsoft Teams access through a separate window. Once
                itâ€™s authorized, you can do more advanced settings on the Teams channel linked with Zeplin.
            </Text>
            <Button
                onClick={(): void => {
                    microsoftTeams.authentication.authenticate({
                        successCallback: onTokenReceive,
                        url: `${BASE_URL}/zeplin/auth/start`
                    });
                }}>
                Log in Zeplin
            </Button>
        </Flex>
    </Flex>
);
