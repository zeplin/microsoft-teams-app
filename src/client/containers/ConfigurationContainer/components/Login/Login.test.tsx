import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { TestProviders } from "../../../../test/TestProviders";
import { Login } from "./Login";

describe("Login", () => {
    it("should render correctly", () => {
        const { container: { firstChild: component } } = render(
            <TestProviders>
                <Login onButtonClick={(): void => undefined}/>
            </TestProviders>
        );

        expect(component).toMatchSnapshot();
    });

    it("should trigger onClick when button is clicked", () => {
        const handler = jest.fn();

        const { getByText } = render(
            <TestProviders>
                <Login onButtonClick={handler}/>
            </TestProviders>
        );

        fireEvent.click(getByText("Log in Zeplin"));

        expect(handler).toBeCalledWith();
    });

    it("should render error row when an error is specified", () => {
        const handler = jest.fn();

        const { container: { firstChild: component } } = render(
            <TestProviders>
                <Login onButtonClick={handler} error="Some error" />
            </TestProviders>
        );

        expect(component).toMatchSnapshot();
    });
});
