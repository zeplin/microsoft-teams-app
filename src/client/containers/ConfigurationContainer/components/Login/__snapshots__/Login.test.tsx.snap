// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div
  class="ui-provider"
  data-uses-unhanded-props="true"
  dir="ltr"
>
  <div
    class="ui-flex"
    data-uses-unhanded-props="true"
  >
    <div />
    <div
      class="ui-flex"
      data-uses-unhanded-props="true"
    >
      <span
        class="ui-text"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        One more step before finishing connection
      </span>
      <span
        class="ui-text"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        Once proceeding, we will ask you to authorize Microsoft Teams access through a separate window. Once it’s authorized, you can do more advanced settings on the Teams channel linked with Zeplin.
      </span>
    </div>
    <div>
      <button
        class="ui-button"
        data-aa-class="Button"
        data-uses-unhanded-props="true"
      >
        <span
          class="ui-button__content"
          data-uses-unhanded-props="true"
          dir="auto"
        >
          Log in Zeplin
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Login should render error row when an error is specified 1`] = `
<div
  class="ui-provider"
  data-uses-unhanded-props="true"
  dir="ltr"
>
  <div
    class="ui-flex"
    data-uses-unhanded-props="true"
  >
    <div />
    <div
      class="ui-flex"
      data-uses-unhanded-props="true"
    >
      <span
        class="ui-text"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        One more step before finishing connection
      </span>
      <span
        class="ui-text"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        Once proceeding, we will ask you to authorize Microsoft Teams access through a separate window. Once it’s authorized, you can do more advanced settings on the Teams channel linked with Zeplin.
      </span>
      <div
        class="ui-flex"
        data-uses-unhanded-props="true"
      >
        <span
          aria-hidden="true"
          class="ui-icon"
          role="img"
        >
          <svg
            focusable="false"
            role="presentation"
            viewBox="8 8 16 16"
          >
            <path
              d="M23.7 20L17.6 9.7c-.1-.2-.3-.4-.5-.5-.2-.1-.4-.2-.7-.2-.2 0-.5.1-.7.2-.2.1-.4.3-.5.5L9.2 20c-.1.2-.2.4-.2.7 0 .4.1.7.4 1 .3.3.6.4 1 .4h12.1c.4 0 .7-.1 1-.4.3-.3.4-.6.4-1 0-.3-.1-.5-.2-.7zM16 12.6c0-.3.2-.5.5-.5s.5.2.5.5v4.7c0 .3-.2.5-.5.5s-.5-.2-.5-.5v-4.7zm.5 8.2c-.4 0-.8-.3-.8-.8 0-.4.3-.8.8-.8s.8.3.8.8c0 .4-.4.8-.8.8z"
            />
          </svg>
        </span>
        <span
          class="ui-text"
          data-uses-unhanded-props="true"
          dir="auto"
        >
          Some error
        </span>
      </div>
    </div>
    <div>
      <button
        class="ui-button"
        data-aa-class="Button"
        data-uses-unhanded-props="true"
      >
        <span
          class="ui-button__content"
          data-uses-unhanded-props="true"
          dir="auto"
        >
          Log in Zeplin
        </span>
      </button>
    </div>
  </div>
</div>
`;
